### PR TITLE
CA-265469: start xapi-nbd from xapi

### PR DESCRIPTION
**Pull request by thomassa**
An interim commit, this uses systemctl to start the xapi-nbd service
unconditionally at the end of the xapi startup sequence.

This means that on first boot, xapi-nbd is started only after the
autogenerated TLS certificate has been created.

### CODE REVIEW

**Comment by thomassa** [comment]
I've not yet tested this, so don't merge it just yet.

File: ocaml/xapi/xapi.ml [hunk 1]
```diff
@@ -680,6 +680,14 @@ let set_stunnel_legacy_db ~__context () =
     Xapi_host.set_stunnel_legacy ~__context legacy_db
   end
 
+(* Consult database to see if this host has any interfaces on networks
```
Comment (gaborigloi) [inline ocaml/xapi/xapi.ml]: We don't do this right now, or did you add this comment for future work?

**Comment by gaborigloi** [comment]
It turns out that the `systemd` library is for implementing daemons, not for controlling them: https://github.com/juergenhoetzel/ocaml-systemd/

**Comment by mseri** [comment]
Why install a systemd service file that depends on xapi to be started is not suitable for this?

**Comment by gaborigloi** [comment]
@mseri because xapi generates a `.pem` file that is used by `xapi-nbd`.

File: ocaml/xapi/xapi.ml (hunk 1 above)
Comment (thomassa) [reply ocaml/xapi/xapi.ml]: The latter, hence the "TODO" part of the comment.

**Comment by thomassa** [comment]
We have a scheduled work-ticket that will subsume this change, so this is just a quick temporary fix to bring the feature branch back to a state that's good for people trying it out and developing software that interacts with the feature.

**Comment by gaborigloi** [comment]
Alternatively we could somehow use systemd to wait for the `.pem` file to be created, but I couldn't find a way of doing this. There are [path units](https://www.freedesktop.org/software/systemd/man/systemd.path.html), but it seems that these just activate a unit when the path changes.
There is also [`ConditionPathExists`](https://www.freedesktop.org/software/systemd/man/systemd.unit.html#ConditionPathExists=), but it seems that the service simply isn't started when the path doesn't exist. What we want is to wait until the file has been created (until a specified timeout).

**Pull request merged**